src/meshable_arena.cc

```diff
@@ -525,10 +525,7 @@ void MeshableArena::finalizeMesh(void *keep, void *remove, size_t sz) {
 
   void *ptr = mmap(remove, sz, HL_MMAP_PROTECTION_MASK, kMapShared | MAP_FIXED, _fd, keepOff * kPageSize);
   hard_assert_msg(ptr != MAP_FAILED, "mesh remap failed: %d", errno);
-  freePhys(remove, sz);
```

Comment:
I'm not sure I follow what the harm is in calling this multiple times (for C twice in your commit message), but I also think I see why moving this above retains correctness.

Reply:
`freePhys` should try to punch a hole + free the underlying physical page, but C should already be pointing towards A, so whats wrong with that physical page being 'freed' twice?